driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -145,6 +146,19 @@ public void initChannel(final SocketChannel ch) throws Exception {
             public void operationComplete(final ChannelFuture future) throws Exception {
                 if (future.isSuccess()) {
                     channel = channelFuture.channel();
+                    channel.closeFuture().addListener(new ChannelFutureListener() {
+                        @Override
+                        public void operationComplete(final ChannelFuture f2) throws Exception {
+                            PendingReader pending = null;
+                            synchronized (this) {
```

Comment:
Access to PendingReader is guarded throughout the stream.  Another thread could get the connection via the connection pool at the same time and try to access it.